### ISSUE TITLE
[unbound-method] Potential false positive on $.getJSON ?

### ISSUE THREAD

**Issue by scottohara** [typescript-eslint/typescript-eslint#1300]
Not sure if this is a false positive or not (it could be my misunderstanding of how the rule works, or how jQuery's `$.getJSON` is typed/implemented).

I'm working in a legacy codebase where jQuery (`$`) is provided as a global.

To help with testing, an ES module wraps & re-exports the global `$.getJSON()` method (so that it can more easily be mocked by jest).

The `export default $.getJSON;` statement triggers `unbound-method`, even though this method (as far as I'm aware) is effectively a static method that does not use `this`?

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": ["error", { "ignoreStatic": true }]
  }
}
```

##### `getJSON.ts`
```TS
/* global $ */
export default $.getJSON;
```

**Expected Result**
No errors?

**Actual Result**
```
getJSON.ts
  2:16  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`  @typescript-eslint/unbound-method
```

**Additional Info**

Is this a legitimate warning, or a false positive?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.10.0` |
| `@typescript-eslint/parser`        | `2.10.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.7.2` |
| `node`                             | `12.12.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
> Is this a legitimate warning, or a false positive?

It's a legitimate warning.

No matter how a (non-arrow) function in JS is defined, it has an implicit `this` context.
It doesn't matter if it's a function you define separately, a function you define as a method, or a function you define as a property - they all have an implicit `this`.

For example:
```ts
function exampleFn() {
  return this.a;
}

const example = {
  a: 1,
  b() { return this.a },
  c: exampleFn,
}

example.b() === 1
example.c() === 1
exampleFn() === undefined // this is implicitly set to the "global" closure (window in the browser)
```

In the case of jQuery, it's a library that (from typescript's perspective) is purely defined in types. So from the POV of the rule - you're returning a reference to a function that is potentially unbound, that may or may not use `this`.

If it turns out the function is unbound, and it does use `this`, then your code will break.
I.e. legitimate warning.

**Comment by scottohara** [comment]
Got it...thanks @bradzacher
